lib/Dialect/Torch/Transforms/RefineTypes.cpp

```diff
@@ -1239,6 +1242,17 @@ ChangeResult TypeAnalyzer::visitAtenShapeAsTensorOp(
   return getLatticeElement(op.getResult()).join(knowledge);
 }
 
+ChangeResult TypeAnalyzer::visitNumToTensorOp(PrimNumToTensorScalarOp op) {
+  auto knowledge =
+      ValueKnowledge::getNotNonePessimisticValueState(op->getContext());
+  fillInSizesGivenSizesList(knowledge, {0});
+
+  knowledge.dtype = IntegerType::get(op.getContext(), 64, IntegerType::Signed);
```

Comment:
The result type might not be integerType.  Using `getDefaultDtypeForTorchScalar` is probably good enough for now.